js/ssai_pulse.js

```diff
@@ -710,6 +748,29 @@ OO.Ads.manager(function(_, $)
       return _appendParamToUrl(url, SMART_PLAYER);
     }, this);
 
+    /**
+     * Parses the ad url to obtain the ssai guid and embed code
+     * @private
+     * @method SsaiPulse#_parseUrl
+     * @param {string} url The stream url
+     */
+    var _parseUrl = _.bind(function(url)
+    {
+      var urlParts = url.split("?");
```

Comment:
Can you add some null checks or a try/catch block?  Just in case someone thinks it's useful and wants to use it for something else.

Reply:
@pilievOoyala added, I think it is safe now but let me know if you see anything still missing

Reply:
If url is null or undefined, an error will still be thrown right? I think we need another null check for "url"

Reply:
@aeng7 fixed

Reply:
If url is undefined, we'll still have an error here. We can probably check typeof url === "string" to cover all use cases so we can safely call .split on the next line.